rr/display/display_manager.py

```diff
@@ -81,15 +80,17 @@ def add_stream(self, key):
         if MAX_STREAMS == len(self._list):
             raise DisplayManagerError("Max number of streams reached")
 
-        if key not in self._list:
-            self._list.append(key)
+        media_name = media.get_name()
+        if media_name not in self._list:
+            self._list.append(media_name)
+            self._appsrc_dict[media_name] = media.get_media()
```

Comment:
This is not an appsrc. Is there even a `media.get_media()?`

Reply:
The appsrc dictionary must be filled when the pipeline is created and you take a reference to the different appsrc in there.